xwiki-platform-core/xwiki-platform-uiextension/xwiki-platform-uiextension-api/src/main/java/org/xwiki/uiextension/internal/WikiUIExtension.java

```diff
@@ -158,35 +177,40 @@ private Object before(boolean inline) throws RenderingException
             throw new RenderingException("Failed to get ui extension document", e);
         }
         Map<String, Object> uixContext = new HashMap<>();
-        uixContext.put(WikiUIExtension.CONTEXT_UIX_DOC_KEY, document.newDocument(xcontext));
-        uixContext.put(WikiUIExtension.CONTEXT_UIX_INLINE_KEY, inline);
+        uixContext.put(CONTEXT_UIX_DOC_KEY, document.newDocument(xcontext));
+        uixContext.put(CONTEXT_UIX_INLINE_KEY, inline);
 
         // Remember the previous uix context to restore it
-        Map<String, Object> previousUIXContext = (Map<String, Object>) xcontext.get(WikiUIExtension.CONTEXT_UIX_KEY);
-        // Put the UIX context in the XWiki context
-        xcontext.put(WikiUIExtension.CONTEXT_UIX_KEY, uixContext);
-
-        return previousUIXContext;
+        this.previousUIXContext = (Map<String, Object>) xcontext.get(CONTEXT_UIX_KEY);
+        // Put the UIX context in the XWiki context. Note that this is deprecated and using the UIX templates is
+        // preferred.
+        xcontext.put(CONTEXT_UIX_KEY, uixContext);
+        // Put the UIX context in the velocity context "uix" key.
+        ScriptContext scriptContext = this.scriptContextManager.getScriptContext();
```

Comment:
Better use getCurrentScriptContext() in your use case.

Reply:
But from the javadoc :)

> In doubt always use {@link #getScriptContext()}.

What are the pros of using `getCurrentScriptContext()` in practice?

Reply:
> In doubt always use {https://github.com/link?type=source #getScriptContext()}.

That does not mean getCurrentScriptContext() is totally useless, and the javadoc also explains that it's for your use case (modifying the context).

> What are the pros of using getCurrentScriptContext() in practice?

As indicated in the javadoc `getScriptContext()` does many things you don't need, so it's better for performances.